integrations/one-drive-personal/syncs/user-files-selection.ts

```diff
@@ -0,0 +1,114 @@
+import { createSync } from 'nango';
+import type { DriveItem } from '../types.js';
+import { toFile } from '../mappers/to-file.js';
+import type { ProxyConfiguration } from 'nango';
+import { OneDriveFileSelection, OneDriveMetadata } from '../models.js';
+
+const sync = createSync({
+    description: "Fetch selected files and folders from a user's OneDrive personal account based on provided metadata.",
+    version: '1.0.0',
+    frequency: 'every hour',
+    autoStart: false,
+    syncType: 'full',
+
+    endpoints: [
+        {
+            method: 'GET',
+            path: '/user-files/selected',
+            group: 'Files'
+        }
+    ],
+
+    scopes: ['OneDrive.ReadWrite', 'offline_access'],
+
+    models: {
+        OneDriveFileSelection: OneDriveFileSelection
+    },
+
+    metadata: OneDriveMetadata,
+
+    exec: async (nango) => {
+        // Personal accounts have a single root drive.
+        const metadata = await nango.getMetadata<OneDriveMetadata>();
+        const fileIds = metadata?.fileIds || [];
+
+        if (fileIds.length === 0) {
+            await nango.log('No fileIds provided in metadata');
+            return;
+        }
+
+        const files: OneDriveFileSelection[] = [];
+        const BATCH_SIZE = 100;
+
+        for (const fileId of fileIds) {
+            const itemConfig: ProxyConfiguration = {
+                //  https://learn.microsoft.com/en-us/graph/api/driveitem-get
+                endpoint: `/v1.0/drive/items/${fileId}`,
+                retries: 10
+            };
+
+            try {
+                const response = await nango.get<DriveItem>(itemConfig);
+                const item = response.data;
+
+                files.push(toFile(item, 'root'));
+
+                if (item.folder && item.folder.childCount > 0) {
+                    await fetchFolderContents(nango, fileId, files);
+                }
+
+                if (files.length >= BATCH_SIZE) {
+                    await nango.batchSave(files, 'OneDriveFileSelection');
+                    files.length = 0;
+                }
+            } catch (error: any) {
```

Comment:
don't think this try catch is necessary, would rather let Nango catch and expose this error